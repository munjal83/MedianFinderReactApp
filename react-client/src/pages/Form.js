import React from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import Paper from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

const styles = {

    textField: {
        marginLeft: 120,
        marginRight: 100,
        width: 200,
    },
    row: {
        display: 'flex',
        justifyContent: 'center',
        margin: '0 auto',
        flexWrap: 'wrap',
        fontSize: '2em'
    },
}

class Form extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            number: '',
            alert: '',
        }

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({ number: event.target.value, alert: '' });        
    }

    handleSubmit(event) {
        if (Number.isInteger(Number(this.state.number)) && this.state.number >= 0) {
            this.props.sendNumber(this.state.number);
            event.preventDefault();
        } else {
            this.setState({ alert: "Please input a valid number" });
        }
    }

    render() {
        
        return (
            <div style={styles.row}>
                <form autoComplete="off">
                    <TextField
                        id="number"
                        label="Please Enter A Number"
                        style={styles.textField}
                        value={this.state.number}
                        onChange={this.handleChange}
                        margin="normal"
                    />
                    <div style={styles.row}>
                        <Button variant="extendedFab" color="primary" onClick={this.handleSubmit}>
                            Get Median
                        </Button>
                    </div>
                </form>
                <div style={styles.row}>
                    <Paper style={styles.root} elevation={1}>
                        <div style={styles.row}>
                        <Typography variant="title" component="h3" color="error">
                            {this.state.alert}
                        </Typography>
                        </div>
                    </Paper>
                </div>
            </div>
        );
    }
}

export default Form;
