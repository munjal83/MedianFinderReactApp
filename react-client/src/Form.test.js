import React from 'react';
import ReactDOM from 'react-dom';
import Form from './pages/Form';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Form />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches the snapshot', () => {
  const tree = renderer.create(<Form />).toJSON();
  expect(tree).toMatchSnapshot();
});