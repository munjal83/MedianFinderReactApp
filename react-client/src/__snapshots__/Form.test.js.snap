// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <form
    autoComplete="off"
  >
    <label
      className="MuiFormLabel-root-92 MuiInputLabel-root-87 MuiInputLabel-animated-91"
    >
      Please Enter A Number
    </label>
    <div
      className="MuiInput-root-99 MuiInput-fullWidth-106 MuiInput-underline-103"
    >
      <input
        className="MuiInput-input-107"
        id="number"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexWrap": "wrap",
          "fontSize": "2em",
          "justifyContent": "center",
          "margin": "0 auto",
          "padding": "5%",
        }
      }
    >
      <button
        className="MuiButtonBase-root-138 MuiButton-root-112 MuiButton-fab-129 MuiButton-extendedFab-130 MuiButton-contained-123 MuiButton-containedPrimary-124 MuiButton-raised-126 MuiButton-raisedPrimary-127"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-113"
        >
          Get Median
        </span>
        <span
          className="MuiTouchRipple-root-141"
        />
      </button>
    </div>
  </form>
  <button
    className="MuiButtonBase-root-138 MuiButton-root-112 MuiButton-text-114 MuiButton-flat-117"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "fontSize": "2em",
        "justifyContent": "center",
        "margin": "0 auto",
        "padding": "5%",
      }
    }
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-113"
    >
      <h3
        className="MuiTypography-root-148 MuiTypography-subheading-155 MuiTypography-colorError-172 MuiTypography-alignCenter-161"
      >
        
      </h3>
    </span>
    <span
      className="MuiTouchRipple-root-141"
    />
  </button>
</div>
`;
